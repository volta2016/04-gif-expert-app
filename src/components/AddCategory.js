import React, { useState } from "react";
import PropTypes from "prop-types";
import btnSearch from "../assets/static/search.svg";

//componente pensado para Form o cuando la persona esccriba algo.

const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState(""); //''por defec -  si lo dejo vacÃ­o es undefined
	const handleInputChange = (e) => {
		setInputValue(e.target.value); //f() que va actualizar el estado
		// console.log('handleInputChange ok');
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		// console.log('Submit Hecho', inputValue);
		if (inputValue.trim().length > 2) {
			setCategories((cate) => [inputValue, ...cate]);
			setInputValue("");
		}
	};

	const handleClear = () => {
		setCategories([]);
	};

	return (
		<form onSubmit={handleSubmit}>
			<p>{inputValue}</p>
			<div className="search">
				<input type="text" value={inputValue} onChange={handleInputChange} />

				<button type="submit" onChange={handleSubmit}>
					<img
						className="search__btn"
						src={`.${btnSearch}`}
						alt="button search"
					/>
				</button>
			</div>
			<button className="search__btn--clear" onClick={handleClear}>
				Clear Gifs
			</button>
		</form>
	);
};

{
	/* va a disparar cada vez que la caja de texto cambie y en onChange.
						dispara un cambio */
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
