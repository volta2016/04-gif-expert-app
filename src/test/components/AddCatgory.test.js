import React from 'react'
import {shallow} from 'enzyme';
import AddCategory from '../../components/AddCategory';


describe('Prueba en el componente <AddCategory />', () => {
  
  const setCategories = () => {}
  const wrapper = shallow(<AddCategory setCategories={ setCategories }/>) 

  test('debe mostrarse correctamente', () => {
      expect(wrapper).toMatchSnapshot();
  });

  test('debe cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', { target: { value: value } });

    expect(wrapper.find('p').text().trim() ).toBe(value);
  })
  
  
});
