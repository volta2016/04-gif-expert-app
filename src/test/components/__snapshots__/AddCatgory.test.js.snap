// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente <AddCategory /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <div
    className="search"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onChange={[Function]}
      type="submit"
    >
      <img
        alt="button search"
        className="search__btn"
        src=".search.svg"
      />
    </button>
  </div>
  <button
    className="search__btn--clear"
    onClick={[Function]}
  >
    Clear Gifs
  </button>
</form>
`;
